### PR TITLE
Enhance performance tests to include git commit loop, config update lo…

### PR DESCRIPTION
**Pull request by rajiesh**
…op and add server monitoring listners

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -15,14 +15,133 @@
...
   File.delete('jmeter.jmx') if File.exists?('jmeter.jmx')
   File.delete('jmeter.log') if File.exists?('jmeter.log')
   File.delete('custom.log') if File.exists?('custom.log')
-  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
   File.delete('perf.jtl') if File.exists?('perf.jtl')
-  ruby "scripts/run_test.rb"
+  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
+end
+
+def scm_commit_loop
```
Comment (zabil) [inline Rakefile]: Pull this out into a common function and pass the command to be executed in the loop with an interval
https://github.com/gocd/support-and-security/blob/37003b221c5c37ef12eb31c4573d04e1fa5c9554/support/analyze#L14

File: Rakefile [hunk 2]
```diff
@@ -15,14 +15,133 @@
...
+      update_config
+      sleep(CONFIG_UPDATE_INTERVAL)
+    end
+  end
+  pid
+end
+
+def extract_files
+  puts "Extract files"
+  `cd #{JMETER_PATH} && unzip apache-jmeter-3.0.zip`
```
Comment (zabil) [inline Rakefile]: use rake cd command,
replace the use of all backticks with either the relevant command or sh

File: scripts/load_scenarios.rb [hunk 3]
```diff
@@ -16,11 +16,148 @@
...
+  (1..NO_OF_PIPELINES).each {|pipeline|
+    url = "#{get_url}/api/admin/pipelines"
+    hash = JSON.parse(File.read('scripts/pipeline.json'))
+    hash["pipeline"]["name"] = "perfpipeline_#{pipeline}"
+    hash["pipeline"]["materials"][0]["attributes"]["url"]  = "git://#{GIT_REPOSITORY_SERVER}/git-repo-#{pipeline}"
+    fJson = File.open("scripts/pipeline.json","w")
+    fJson.write(hash.to_json)
+    fJson.close
+    puts 'create a pipeline'
+    sh(%Q{curl -sL -w "%{http_code}" -X POST  -H "Accept: application/vnd.go.cd.v1+json" -H "Content-Type: application/json" --data "@scripts/pipeline.json" #{url} -o /dev/null})
```
Comment (zabil) [inline scripts/load_scenarios.rb]: Can we use something thats cross platform here?
I am sure we can use ruby methods to make this call

File: scripts/load_scenarios.rb [hunk 4]
```diff
@@ -16,11 +16,148 @@
...
+end
+
+def create_agents
+  json = JSON.parse(open(RELEASES_JSON_URL).read)
+  version, release = json.sort {|a, b| a['go_full_version'] <=> b['go_full_version']}.last['go_full_version'].split('-')
+  go_full_version = "#{version}-#{release}"
+
+  sh('mkdir go-agents')
+  sh(%Q{wget -O go-agents/go-agent-#{go_full_version}.zip https://download.go.cd/experimental/binaries/#{go_full_version}/generic/go-agent-#{go_full_version}.zip})
+  sh("unzip go-agents/go-agent-#{go_full_version}.zip -d go-agents/")
```
Comment (zabil) [inline scripts/load_scenarios.rb]: Again use ruby cross platform libraries to do this.

File: Rakefile [hunk 5]
```diff
@@ -15,14 +15,133 @@
...
+end
+
+def cleanup
   File.delete('jmeter.jmx') if File.exists?('jmeter.jmx')
   File.delete('jmeter.log') if File.exists?('jmeter.log')
   File.delete('custom.log') if File.exists?('custom.log')
-  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
   File.delete('perf.jtl') if File.exists?('perf.jtl')
-  ruby "scripts/run_test.rb"
+  File.delete('jmeter.jtl') if File.exists?('jmeter.jtl')
```
Comment (ketan) [inline Rakefile]: this can just be `rm_rf` for all files.

File: scripts/load_scenarios.rb [hunk 6]
```diff
@@ -16,11 +16,148 @@
...
+      break if agents.size == NO_OF_AGENTS
+    end
+  end
+  sleep(180)
+end
+
+
+def destroy pid
+  p "destroying child process #{pid}"
+  Process.kill 9, pid
```
Comment (ketan) [inline scripts/load_scenarios.rb]: I'd probably execute `kill -3` followed by a wait followed by `kill -9`

**Pull request merged**